x-pack/plugins/stack_connectors/public/connector_types/webhook/webhook_connectors.tsx

```diff
@@ -28,29 +31,207 @@ import {
+
+  const hasInitialCA = !!getFieldDefaultValue<boolean | undefined>('config.ca');
+
+  useEffect(
+    () => setFieldValue('config.hasAuth', authType !== WebhookAuthType.None),
+    [authType, setFieldValue]
+  );
+
+  const basicAuthFields = (
+    <EuiFlexGroup justifyContent="spaceBetween">
+      <EuiFlexItem>
+        <UseField
+          path="secrets.user"
+          config={{
+            label: i18n.USERNAME_LABEL,
+            validations: [
+              {
+                validator: emptyField(i18n.USERNAME_REQUIRED),
+              },
+            ],
+          }}
+          component={Field}
+          componentProps={{
+            euiFieldProps: {
+              readOnly,
+              'data-test-subj': 'webhookUserInput',
+              fullWidth: true,
+            },
+          }}
+        />
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <UseField
+          path="secrets.password"
+          config={{
+            label: i18n.PASSWORD_LABEL,
+            validations: [
+              {
+                validator: emptyField(i18n.PASSWORD_REQUIRED),
+              },
+            ],
+          }}
+          component={PasswordField}
+          componentProps={{
+            euiFieldProps: {
+              'data-test-subj': 'webhookPasswordInput',
+              readOnly,
+            },
+          }}
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+
+  const sslCertAuthFields = (
+    <EuiFlexGroup justifyContent="spaceBetween">
+      <EuiFlexItem>
+        <UseField
+          path="secrets.password"
+          config={{
+            label: i18n.PASSPHRASE_LABEL,
+          }}
+          component={PasswordField}
+          componentProps={{
+            euiFieldProps: {
+              'data-test-subj': 'webhookSSLPassphraseInput',
+              readOnly,
+            },
+          }}
+        />
+        <EuiSpacer size="s" />
+        <UseField
+          path="config.certType"
+          defaultValue={certTypeDefaultValue}
+          component={({ field }) => (
+            <EuiTabs size="s">
+              <EuiTab
+                onClick={() => field.setValue(SSLCertType.CRT)}
+                isSelected={field.value === SSLCertType.CRT}
+              >
+                {i18n.CERT_TYPE_CRT_KEY}
+              </EuiTab>
+              <EuiTab
+                onClick={() => field.setValue(SSLCertType.PFX)}
+                isSelected={field.value === SSLCertType.PFX}
+              >
+                {i18n.CERT_TYPE_PFX}
+              </EuiTab>
+            </EuiTabs>
+          )}
+        />
+        <EuiSpacer size="s" />
+        {certType === SSLCertType.CRT && (
+          <EuiFlexGroup>
+            <EuiFlexItem>
+              <UseField
+                path="secrets.crt"
+                config={{
+                  label: 'CRT file',
+                  validations: [
+                    {
+                      validator: emptyField(i18n.CRT_REQUIRED),
```

Comment:
Should we validate (both on the FE and BE) that the file uploaded were actually `CRT/KEY/PFX` files, perhaps restrict the file format at the least if the intention is to let it fail in the executor?

Reply:
+1

Reply:
I can restrict file extensions in the file picker. PFX files are binary encoded and I don't know an easy way of validating them. Certs will I THINK always contain the string `-----BEGIN CERTIFICATE----` whereas the keys I've generated always contain `-----BEGIN PRIVATE KEY-----` but I think `-----BEGIN RSA PRIVATE KEY-----` is also valid?

I don't have enough knowledge of what constitutes a valid cert or key file so I don't feel comfortable writing a validation for it beyond file extensions.

Reply:
Make sense @Zacqary. Restricting only the file extensions should be enough.